### DISCUSSION TITLE
Give JSON stronger support

### DISCUSSION THREAD

**Discussion by colorworlds** [godotengine/godot-proposals#4059] [Scripting]
At present, gdscript's support for JSON seems incomplete. 

For example, the built-in type of a `Vector2` cannot be converted to an `Array` or `Dictionary`, and conversely, an `Array` or `Dictionary` can not be parsed into a vector through built-in methods.

And if it is possible, I hope godot can support JSON conversion of custom `Object` types. 

Like `object2dic` and `dict2object` in Python.

**Comment by Xrayez** [comment]
See my closed pull request (I can reopen it if needed):

- godotengine/godot#33241

But I think that Godot core developers are not really that interested in enhancing JSON support (or serialization/deserialization for that matter).

Perhaps [JSON5](https://json5.org/) could be incorporated in https://github.com/goostengine/goost, at least.

> Like `object2dic` and `dict2object` in Python.

Are you aware of `dict2inst()` and `inst2dict()` functions in GDScript?

Those could also be improved, though:

- godotengine/godot#33137

**Reply by colorworlds** [reply]
Thank you very much for your previous work.  
I haven't noticed `dict2inst()` and `inst2dict()` before.  Thanks for reminding. These function is very useful to me.

**Comment by Calinou** [comment]
Consider using [ConfigFile](https://docs.godotengine.org/en/latest/classes/class_configfile.html) instead of JSON to save user configuration and other data *that does not need to interoperate with other (non-Godot) software*. ConfigFile can serialize any Godot datatype automatically, without requiring you to write custom code for it. ConfigFile also happens to be more human-readable (and even writable) overall.

Supporting the serialization of Godot datatypes in JSON wouldn't be very useful for the goal of interoperability I mentioned, since other software won't automatically know how to interpret those datatypes.
